### PR TITLE
Additional documentation and renames

### PR DESCRIPTION
**Pull request by kerzhner**
Follows https://github.com/statechannels/dispute-game/pull/46.

Role is renamed to identity as the dispute game no longer has roles. Any (valid) participant should be able to initiate the dispute game to challenge a claim by some other participant.

### CODE REVIEW

File: src/challenge-manager.ts [hunk 1]
```diff
@@ -68,7 +68,26 @@ export function expectedNumOfLeaves(
     : disputedStep - consensusStep + 1;
 }
 
-// When implemented in Solidity, the challenger will deploy the contract
+/**
+ * ChallengeManager prototypes the contract to be deployed on-chain.
```
Comment (lalexgap) [inline src/challenge-manager.ts:72]: This is great documentation 🎉

File: src/tests/challenge-manager.test.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import {ChallengeManager, State, stepForIndex} from '../challenge-manager';
 import {DisputeGame} from '../dispute-game';
 import {generateWitness, WitnessProof} from '../merkle';
 
-export type Role = 'challenger' | 'proposer';
+export type Identity = 'challenger' | 'proposer';
```
Comment (lalexgap) [inline src/tests/challenge-manager.test.ts:8]: Should this belong in the challenger-agent file instead?

**Review by lalexgap — APPROVED** [review]
👍

File: src/tests/challenge-manager.test.ts (hunk 2 above)
Comment (kerzhner) [reply src/tests/challenge-manager.test.ts]: That's a good call! be3b0f8

**Pull request merged**